Support downloading password protected files with 1Fichier backend

hkr opened the pull request:
#### What is the purpose of this change?

Add a new advanced option `shared_password` to support accessing files in shared folders that are protected by a password.

#### Was the change discussed in an issue or in the forum before?

https://github.com/rclone/rclone/issues/3353

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [ ] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

ncw commented on backend/fichier/fichier.go:
```diff
@@ -44,6 +44,11 @@ func init() {
+			Help:     "If you want to download a shared file that is password protected, add this parameter",
+			Name:     "shared_password",
+			Required: false,
+			Advanced: true,
```
This should probably say `IsPassword: true,` here - we are entering a password aren't we. This will mean it is stored obfuscated in the config and will need to be revealed.

ncw commented on backend/fichier/api.go:
```diff
@@ -90,6 +90,7 @@ func (f *Fs) getDownloadToken(ctx context.Context, url string) (*GetTokenRespons
 	request := DownloadRequest{
 		URL:    url,
 		Single: 1,
+		Pass: f.opt.SharedPassword,
```
This will be passing empty string when SharedPassword is not set - is that OK?

hkr replied:
I didn't have any problems with that.
However I made another change and added `omitempty`.
I have not tried to set a password using the API but in the web UI an empty string removes the password.

ncw reviewed (commented):
This looks useful thanks :-)

Can you drop the doc changes from the commit please? So rclone.1 MANUAL.* - these get made as part of the release.

ncw commented:
@buengese any thoughts on this?

hkr commented:
I noticed that 1Fichier also has folder passwords. Currently 'list' does not work on those.
Any opinion if I should use the same `shared_password` option also for folder password or is it better to add a new option?

buengese commented:
This looks mostly good now.
Regarding folder passwords: In theory file and folder password are independent, so this should be a different option. In reality, I don't think I've ever actually seen a folder password, and I'm not even sure where you would set them.

hkr commented:
https://github.com/hkr/rclone/commit/30facdac89a1f3b8c4809cb0d606b7d4fd4fea34 worked for me for folder password. It's not part of this PR

ivandeex commented:
Code quality test failed. Please run `gofmt -w ./backend/fichier/*.go` and commit the result.

hkr commented:
> This looks mostly good now.
> Regarding folder passwords: In theory file and folder password are independent, so this should be a different option. In reality, I don't think I've ever actually seen a folder password, and I'm not even sure where you would set them.

You can set them in the Web UI similar to file password. Click a folder and then click the lock icon.

buengese commented:
>You can set them in the Web UI similar to file password. Click a folder and then click the lock icon.
Thanks. That's what I meant. I had so far missed that option. Since it's possible to have files with a different password in a password-protected folder it' would make sense to have separate parameters for file and folder passwords.

hkr commented:
I renamed `shared_password` to `file_password`, merged the changes to list files of a password protected folder and added a new option `folder_password`

buengese commented:
I've merged this now sorry for the delay.

ivandeex commented:
Merged by @buengese in https://github.com/rclone/rclone/commit/92b3518c78848d454a7352707fa7c849fd93403a
